sys/kern/src/kipc.rs

```diff
@@ -507,3 +510,68 @@ fn find_faulted_task(
         .set_send_response_and_length(0, response_len);
     Ok(NextTask::Same)
 }
+
+fn read_panic_message(
+    tasks: &mut [Task],
+    caller: usize,
+    message: USlice<u8>,
+    response: USlice<u8>,
+) -> Result<NextTask, UserError> {
+    let index: u32 = deserialize_message(&tasks[caller], message)?;
+    let index = index as usize;
+    let Some(task) = tasks.get(index) else {
+        return Err(UserError::Unrecoverable(FaultInfo::SyscallUsage(
+            UsageError::TaskOutOfRange,
+        )));
+    };
+
+    // Make sure the task is actually panicked.
+    let TaskState::Faulted {
+        fault: FaultInfo::Panic,
+        ..
+    } = task.state()
+    else {
```

Comment:
huuuh, I've never seen a `let`/`else` used in a way that doesn't bind any names like this. I assume you prefer this over `if matches!`?

Reply:
Yeah, it just felt a little bit better than 
```rust
if !matches!(task.state(), TaskState::Faulted { fault: FaultInfo::Panic, .. })) {
    return ...;
}
```
for some vague aesthetic reason. But it should be equivalent.